dsp/window/cmd/leakage/leakage.go

```diff
@@ -0,0 +1,387 @@
+	})
+	if err != nil {
+		log.Fatalf("ΔF_0: %v", err)
+	}
+	deltaF0Line.Color = color.RGBA{R: 0xff, A: 0xff}
+
+	deltaFhalfLine, err := plotter.NewLine(plotter.XYs{
+		{X: c.deltaFhalf / 2, Y: 0}, {X: c.deltaFhalf / 2, Y: min},
+	})
+	if err != nil {
+		log.Fatalf("ΔF_0.5: %v", err)
+	}
+	deltaFhalfLine.Color = color.RGBA{G: 0xff, A: 0xff}
+
+	var blank plotter.Line
+
+	p.Add(
+		gammaLine,
+		deltaF0Line,
+		deltaFhalfLine,
+		spectrumLines,
+	)
+	p.Legend.Add(fmt.Sprintf("ΔF_0 = %.3v", c.deltaF0), deltaF0Line)
+	p.Legend.Add(fmt.Sprintf("ΔF_0.5 = %.3v", c.deltaFhalf), deltaFhalfLine)
+	p.Legend.Add(fmt.Sprintf("K = %.3v", c.k()), &blank)
+	p.Legend.Add(fmt.Sprintf("ɣ_max = %.3v", c.gammaMax), gammaLine)
+	p.Legend.Add(fmt.Sprintf("β = %.3v", c.beta), &blank)
+	p.Legend.Top = true
+	p.Legend.XOffs = -10
+	p.Legend.YOffs = -10
+
+	err = p.Save(vg.Length(*width)*vg.Centimeter, vg.Length(*height)*vg.Centimeter, *out)
+	if err != nil {
+		log.Fatal(err)
+	}
+}
+
+// characteristics hold DFT window characteristic statistics.
+// See http://www.dsplib.ru/content/win/win.html for details.
+type characteristics struct {
+	deltaF0    float64
+	deltaFhalf float64
+	gammaMax   float64
+	beta       float64
+}
+
+// k returns the K window parameter which is the ratio of the window's
+// ΔF_0 to the ΔF_0 of the rectangular window.
+func (c *characteristics) k() float64 {
+	return c.deltaF0 / 2
+}
+
+func funcCharacteristics(fn func([]float64) []float64, n, m int, symm bool) (c *characteristics, xy plotter.XYs, min float64, err error) {
+	if m < n {
+		return nil, nil, 0, fmt.Errorf("window: sequence too short for window: %d < %d", m, n)
+	}
+
+	var w []float64
+	t := make([]float64, m)
+	if symm {
+		w = window.NewValues(fn, n)
+	} else {
+		w = window.NewValues(fn, n+1)[:n]
+	}
+
+	copy(t, w)
+
+	var max float64
+	xy = make(plotter.XYs, m/2+1)
+	fft := fourier.NewFFT(len(t))
+	for i, c := range fft.Coefficients(nil, t) {
+		a := db(cmplx.Abs(c))
+		t[i] = a
+
+		if math.IsInf(a, -1) {
+			a = min
+		} else if a < min {
+			min = a
+		}
+		if i == 0 {
+			max = a
+		}
+		xy[i] = plotter.XY{X: float64(i) * float64(n) / float64(m), Y: a - max}
+	}
+
+	c = &characteristics{beta: db(stat.Mean(w, nil))}
+	c.deltaF0, c.deltaFhalf, c.gammaMax = parameters(t, n, m)
+
+	return c, xy, min - max, nil
+}
+
+func csvCharacteristics(r io.Reader, n, m int) (c *characteristics, xy plotter.XYs, min float64, err error) {
+	if m < n {
+		return nil, nil, 0, fmt.Errorf("window: sequence too short for window: %d < %d", m, n)
+	}
+	sc := bufio.NewScanner(r)
+	max := math.Inf(-1)
+	var t []float64
+	for sc.Scan() {
+		text := sc.Text()
+		if strings.HasPrefix(text, "#") {
+			continue
+		}
+		v, err := strconv.ParseFloat(text, 64)
+		if err != nil {
+			log.Fatal(err)
+		}
+		if v > max {
+			max = v
+		}
+		t = append(t, v)
+	}
+
+	xy = make(plotter.XYs, len(t))
+	for i, a := range t {
+		if math.IsInf(a, -1) {
+			a = min
+		} else if a < min {
+			min = a
+		}
```

Comment:
I vaguely remember the Go style recommending to use `switch` for this case.
(feel free to ignore, though)
```suggestion
		switch {
		case math.IsInf(a, -1):
			a = min
		case if a < min:
			min = a
		}
```

Reply:
Looking more carefully at this, the logic is not correct; it works approximately right for most cases but can deviate from the intention with some inputs.